crates/uv-resolver/src/pubgrub/dependencies.rs

```diff
@@ -221,21 +218,65 @@ fn to_pubgrub(
                 version,
             ))
         }
+        RequirementSource::Url { url, subdirectory } => {
+            let mut url: Url = url.deref().clone();
+            if let Some(subdirectory) = subdirectory {
+                url.set_fragment(Some(&format!("subdirectory={}", subdirectory.display())));
+            }
+            let Some(expected) = urls.get(&requirement.name) else {
+                return Err(ResolveError::DisallowedUrl(
+                    requirement.name.clone(),
+                    url.to_string(),
+                ));
+            };
+
+            if !Urls::is_allowed(expected, &VerbatimUrl::from_url(url.clone())) {
+                return Err(ResolveError::ConflictingUrlsTransitive(
+                    requirement.name.clone(),
+                    expected.verbatim().to_string(),
+                    url.to_string(),
+                ));
+            }
```

Comment:
Hmm, I'm not a big fan of what's happening in this branch... I don't like that we're having to recreate these URLs all over the place. Also don't like how much `VerbatimUrl` conversion and URL cloning and deref and such that we're needing to do. What do you see as the better / correct solution here?

Reply:
I've front-loaded this by putting a `VerbatimUrl` with the joined url on the url variants of `RequirementSource`.